### PR TITLE
supervisor namespace and delete trainee

### PR DESCRIPTION
**Pull request by MuhammadTamzid**

### CODE REVIEW

File: app/assets/javascripts/supervisor/users.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (chuanhtuan) [inline app/assets/javascripts/supervisor/users.coffee]: remove unused file

File: app/assets/stylesheets/supervisor/users.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the supervisor/Users controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/supervisor/users.scss]: remove unused file

File: app/controllers/supervisor/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    @user.attributes = {role: "trainee"}
+    if @user.save
+      flash[:success] = t(:add_trainee_sucessful_message)
+      redirect_to supervisor_users_path(role: "trainee")
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: should define a constant in model
User::ROLE[:trainee]  will return "trainee"

File: app/controllers/supervisor/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end
+
+  def edit
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t(:delete_trainee_sucessful_message)
+    redirect_to supervisor_users_path(role: "trainee")
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: above

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -38,6 +24,12 @@ def update
     end
   end
 
+  def destroy
+    User.find(params[:id]).destroy
+    flash[:success] = t(:delete_trainee_sucessful_message)
+    redirect_to users_url(role: "trainee")
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: above

File: app/views/supervisor/users/_user.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<li>
+  <%= gravatar_for user, size: 50 %>
+  <%= link_to user.name, user %>
+  <% if current_user.supervisor? && !current_user?(user) && !user.supervisor? %>
+    | <%= link_to "delete", user, method: :delete,
+                                  data: { confirm: "You sure?" } %>
```
Comment (chuanhtuan) [inline app/views/supervisor/users/_user.html.erb]: i18n

File: config/routes.rb [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
+    root                'static_pages#home'
+    get    'help'    => 'static_pages#help'
+    get    'about'   => 'static_pages#about'
+    get    'contact' => 'static_pages#contact'
+    get    'login'   => 'sessions#new'
+    post   'login'   => 'sessions#create'
+    delete 'logout'  => 'sessions#destroy'
+
+    namespace :supervisor do
+      root            'static_pages#home'
```
Comment (chuanhtuan) [inline config/routes.rb]: space

File: app/controllers/supervisor/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    @user = User.find params[:id]
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+    @user.attributes = {role: "trainee"}
```
Comment (chuanhtuan) [inline app/controllers/supervisor/users_controller.rb]: use constant

File: app/helpers/supervisor/users_helper.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+module Supervisor::UsersHelper
```
Comment (chuanhtuan) [inline app/helpers/supervisor/users_helper.rb]: remove unused file

File: app/controllers/supervisor/users_controller.rb (hunk 8 above)
Comment (MuhammadTamzid) [inline app/controllers/supervisor/users_controller.rb]: Sorry Sir :(. We should fixed it in our previous commit.

**Pull request merged**